PyPDF2/_writer.py

```diff
@@ -794,6 +809,20 @@ def write(self, stream: StreamType) -> None:
         self._write_trailer(stream)
         stream.write(b_(f"\nstartxref\n{xref_location}\n%%EOF\n"))  # eof
 
+    def write(self, fileobj: StrByteType) -> None:
+        my_file = False
+
+        if isinstance(fileobj, str) and fileobj:
```

Comment:
I don't think having `fileobj` be optional would work, and so can drop the `and fileobj`:

```suggestion
        if isinstance(fileobj, str):
```